packages/node-cli/src/commands/UploadFilesCommandGenerator.js

```diff
@@ -0,0 +1,188 @@
+			if (!pathIsUnrestricted(name)) {
+				disabledMessage = TranslationService.getMessage(MESSAGES.RESTRICTED_FOLDER);
+			} else if (!this._fileSystemService.getFilesFromDirectory(folder).length) {
+				disabledMessage = TranslationService.getMessage(MESSAGES.EMPTY_FOLDER);
+			}
+
+			return {
+				name: name,
+				value: folder,
+				disabled: disabledMessage,
+			};
+		};
+
+		const localFileCabinetFoldersChoices = localFileCabinetSubFolders.map(transformFoldersToChoicesFunc);
+
+		return [
+			{
+				message: TranslationService.getMessage(QUESTIONS.SELECT_FOLDER),
+				type: CommandUtils.INQUIRER_TYPES.LIST,
+				name: COMMAND_ANSWERS.SELECTED_FOLDER,
+				choices: localFileCabinetFoldersChoices,
+			},
+		];
+	}
+
+	_generateSelectFilesQuestion(selectedFolder) {
+		const files = this._fileSystemService.getFilesFromDirectory(selectedFolder);
+
+		const transformFilesToChoicesFunc = file => {
+			const path = this._getFileCabinetRelativePath(file);
+			return { name: path, value: path };
+		};
+		const filesChoices = files.map(transformFilesToChoicesFunc);
+
+		return [
+			{
+				message: TranslationService.getMessage(QUESTIONS.SELECT_FILES),
+				type: CommandUtils.INQUIRER_TYPES.CHECKBOX,
+				name: COMMAND_OPTIONS.PATHS,
+				choices: filesChoices,
+				validate: fieldValue => showValidationResults(fieldValue, validateArrayIsNotEmpty),
+			},
+		];
+	}
+
+	_getFileCabinetRelativePath(file) {
+		return file.replace(this.localFileCabinetFolder, '').replace(/\\/g, '/');
+	}
+
+	_generateOverwriteQuestion() {
+		return {
+			type: CommandUtils.INQUIRER_TYPES.LIST,
+			name: COMMAND_ANSWERS.OVERWRITE_FILES,
+			message: TranslationService.getMessage(QUESTIONS.OVERWRITE_FILES),
+			default: 0,
+			choices: [
+				{ name: TranslationService.getMessage(YES), value: true },
+				{ name: TranslationService.getMessage(NO), value: false },
+			],
+		};
+	}
+
+	_preExecuteAction(answers) {
+		const { PROJECT, PATHS } = COMMAND_OPTIONS;
+		answers[PROJECT] = CommandUtils.quoteString(this._projectFolder);
+		if (answers.hasOwnProperty(PATHS)) {
+			if (Array.isArray(answers[PATHS])) {
+				answers[PATHS] = answers[PATHS].map(CommandUtils.quoteString).join(' ');
+			} else {
+				answers[PATHS] = CommandUtils.quoteString(answers[PATHS]);
+			}
+		}
+		return answers;
+	}
+
+	_executeAction(answers) {
+		const executionContextUploadFiles = new SDKExecutionContext({
+			command: this._commandMetadata.sdkCommand,
+			includeProjectDefaultAuthId: true,
+			params: answers,
+		});
+
+		return executeWithSpinner({
+			action: this._sdkExecutor.execute(executionContextUploadFiles),
+			message: TranslationService.getMessage(MESSAGES.UPLOADING_FILES),
+		});
+	}
+
+	_formatOutput(operationResult) {
+		const { data } = operationResult;
+
+		if (SDKOperationResultUtils.hasErrors(operationResult)) {
+			SDKOperationResultUtils.logResultMessage(operationResult);
+			SDKOperationResultUtils.logErrors(operationResult);
+			return;
+		}
+
+		if (Array.isArray(data)) {
```

Comment:
There may not be a need to check if data is an array. I think from the operation result, it'll will always return data as an array. I'm not 100% sure though, please check.

Reply:
This is to protect from the possibility that it could be null.

Reply:
But in this case, I think we know it'll always be an array right? But I'm not 100% sure so I'm ok if this check is necessary here.